cmake/llext-edk.cmake

```diff
@@ -137,54 +138,123 @@ foreach(dir ${include_dirs})
+    set(${str_out} "${str_escaped}" PARENT_SCOPE)
+endfunction()
 
-list(JOIN imacros_gen_make " " imacros_gen_str)
-file(APPEND ${llext_edk}/Makefile.cflags "\n\nLLEXT_GENERATED_IMACROS_CFLAGS = ${imacros_gen_str}")
+# Clear the contents of the requested file
+function(edk_write_header target)
+    file(WRITE ${edk_file_${target}} "")
+endfunction()
 
-# Generate flags for CMake
-list(APPEND all_flags_cmake ${base_flags_cmake} ${imacros_gen_cmake} ${all_inc_flags_cmake})
-file(WRITE ${llext_edk}/cmake.cflags "set(LLEXT_CFLAGS ${all_flags_cmake})")
+# Mark a section in the file with a single line comment
+function(edk_write_comment target comment)
+    file(APPEND ${edk_file_${target}} "\n# ${comment}\n")
+endfunction()
 
-file(APPEND ${llext_edk}/cmake.cflags "\n\nset(LLEXT_ALL_INCLUDE_CFLAGS ${all_inc_flags_cmake})")
+# Define a variable in the file
+function(edk_write_var target var_name var_value)
+    if(target STREQUAL "CMAKE")
+        # CMake: export assignments of the form:
+        #
+        #   set(var "value1;value2;...")
+        #
+        set(DASHIMACROS "-imacros\${CMAKE_CURRENT_LIST_DIR}/")
+        set(DASHI "-I\${CMAKE_CURRENT_LIST_DIR}/")
+        string(CONFIGURE "${var_value}" exp_var_value @ONLY)
+        # Embedded quotes and backslashes are escaped. The list is otherwise
+        # exported verbatim, surrounded by quotes.
+        edk_escape("${exp_var_value}" exp_var_value)
+        file(APPEND ${edk_file_${target}} "set(${var_name} \"${exp_var_value}\")\n")
+    elseif(target STREQUAL "MAKEFILE")
+        # Makefile: export assignments of the form:
+        #
+        #   var = "value1" "value2" ...
+        #
+        set(DASHIMACROS "-imacros\$(${install_dir_var})/")
+        set(DASHI "-I\$(${install_dir_var})/")
+        string(CONFIGURE "${var_value}" exp_var_value @ONLY)
+        # Embedded quotes and backslashes are escaped. Each element of the
+        # list is wrapped in quotes and is separated by a space.
+        edk_escape("${exp_var_value}" exp_var_value)
+        list(JOIN exp_var_value "\" \"" exp_var_value_str)
+        file(APPEND ${edk_file_${target}} "${var_name} = \"${exp_var_value_str}\"\n")
+    elseif(target STREQUAL "BASH")
+        # Bash: export array variables of the form:
+        #
+        #   var=("value1" "value2" ...)
+        #
+        # TIP: use the "${var[@]}" syntax, including quotes, to pass all the
+        # elements as single arguments to a command in scripts.
+        set(DASHIMACROS "-imacros\${${install_dir_var}}/")
+        set(DASHI "-I\${${install_dir_var}}/")
+        string(CONFIGURE "${var_value}" exp_var_value @ONLY)
+        # Embedded quotes and backslashes are escaped. Each element of the
+        # list is wrapped in quotes and is separated by a space.
+        edk_escape("${exp_var_value}" exp_var_value)
+        list(JOIN exp_var_value "\" \"" exp_var_value_str)
+        file(APPEND ${edk_file_${target}} "${var_name}=(\"${exp_var_value_str}\")\n")
+    endif()
```

Comment:
I don't think we should start exporting to specific shells.
This is a build system, and thus exporting CMake and Makefiles seems ok, but starting to export to bash is going to just bloat our interface.

What about `bat` files ? `fish` ? or the favorite shell of some other user ?
If an external build tool is not satisfied with CMake or Makefiles then I would leave it for that tool to have a tool which can read the file and convert to its favorite shell.

Only additional file format I consider acceptable is yaml, because that is a file format Zephyr build system already uses for dumping data, for example in the Zephyr meta file creation.

Reply:
> I'm surprised this would be considered "bloat" 

when I'm saying bloat then i'm thinking in terms of what we promise externally.
If we provide bash here, should we then also provide this elsewhere, and what about xyz.

> And to think I was about to add Python export as well...

had you done it the other way, then a Python might have made it through :laughing: 
because we already use Python a lot, and Python is cross-platform (Linux, Windows, MacOS, etc.), but with Python then a yaml file would have been just as easy / nice to use :wink: 
> Only additional file format I consider acceptable is yaml